### PR TITLE
Add ETF module to monorepo

### PR DESCRIPTION
**Pull request by adairrr**
This change migrates the ETF app from the old repository (https://github.com/AbstractSDK/apps/tree/main/contracts) to here

### CODE REVIEW

File: modules/contracts/apps/etf/src/state.rs
Comment (Buckram123) [inline modules/contracts/apps/etf/src/state.rs:14]: Would be nice to add comment why we start state key with NULL

**Review by Buckram123 — CHANGES_REQUESTED** [review]
Happy to finally see this contract! :rofl:

File: modules/contracts/apps/etf/rustfmt.toml
Comment (adairrr) [reply modules/contracts/apps/etf/rustfmt.toml:14]: ha! Will remove these since they're just leftover from the previous package.

File: modules/contracts/apps/etf/src/handlers/instantiate.rs
Comment (adairrr) [reply modules/contracts/apps/etf/src/handlers/instantiate.rs:22]: That would certainly be best practice. Thanks for pointing it out.

Comment (adairrr) [reply modules/contracts/apps/etf/src/handlers/instantiate.rs:22]: We should honestly convert this over to using token-factory anyway...

Comment (adairrr) [reply modules/contracts/apps/etf/src/handlers/instantiate.rs:22]: Perhaps we should wait until #36 is merged and then use it as a usage example?

File: modules/contracts/apps/etf/src/state.rs
Comment (adairrr) [reply modules/contracts/apps/etf/src/state.rs:14]: Is it common practice to do so? I haven't seen that elsewhere 👀

File: modules/contracts/apps/etf/src/handlers/instantiate.rs
Comment (adairrr) [reply modules/contracts/apps/etf/src/handlers/instantiate.rs:22]: > > We should honestly convert this over to using token-factory anyway...
> 
> Is token factory supported on every chain we use?

Not Archway :/

File: modules/contracts/apps/etf/src/state.rs
Comment (Buckram123) [reply modules/contracts/apps/etf/src/state.rs:14]: > Is it common practice to do so? I haven't seen that elsewhere eyes

Same, haha. Saw just random bytes before(to decrease key length), but not a normal key starting with NULL

Comment (adairrr) [reply modules/contracts/apps/etf/src/state.rs:14]: Ah, yeah that used to be required in CW denoting how long the key was. state = 5 chars, fee = 3 chars.

Comment (Buckram123) [reply modules/contracts/apps/etf/src/state.rs:14]: Oh wow, how old is this contract

Comment (adairrr) [reply modules/contracts/apps/etf/src/state.rs:14]: > Oh wow, how old is this contract

Howard wrote it as one of the first Abstract modules >1 year ago.

**Comment by CyberHoward** [comment]
Lets merge this as an example contract for now and leave it as it is until there's a use-case for it.

**Review by cyberhoward — APPROVED** [review]

**Pull request merged**
